### PR TITLE
ref(bulk-deletes): bulk delete query path and producer

### PR DESCRIPTION
**Pull request by MeredithAnya**
**context**
The current pipeline we have for deleting data (via lightweight deletes) is an API endpoint that executes the query directly. `DELETE` requests are technically synchronous at the moment due to the ClickHouse version we are on (the `lightweight_delete_mutations_sync` parameter exist in newer version).

**problem**
We'd like to keep the number of concurrent `DELETE` requests to a minimum since they add an increased load to ClickHouse. Though the number of rows being deleted does have an effect on load, even a delete that deletes a few rows can have a significant enough effect because it's still a mutation (under the hood the `DELETE FROM` becomes an `UPDATE`). 

**batched deletes**
In order to scale we are adding a consumer that will batch up DELETE queries so that we have fewer mutations, even though the number of rows deleted/mutated will be higher.

The new pipeline is a bit different and has some new components:
- [ ] `DELETE` api endpoint will now take a request and produce it to a topic (This PR)
    * The topics are storage specific, ex. `LW_DELETIONS_SEARCH_ISSUES = "snuba-lw-deletions-search-issues"`
- [ ] Consumer(s) will be added to consume from the topic that corresponds to the storage the consumer is for, right now this will just be search issues

**other notable changes (in this pr)**
* addition of `allowed_columns` in the deletion settings, this will eventually replace the `ColumnFilterProcessor` once deletes are fully on the bulk delete endpoint
* `rows_to_delete` is added to the delete query payload so that we can batch based on number of rows to be delete instead of batching on number of messages

### CODE REVIEW

File: snuba/datasets/deletion_settings.py
Comment (armenzg) [inline snuba/datasets/deletion_settings.py:6]: What happens if an issue has more than these many rows?

File: snuba/web/bulk_delete_query.py
Comment (onkar) [inline snuba/web/bulk_delete_query.py:110]: To avoid sorting lists explicitly, can we do this?

```
difference =  set(conditions.keys()) - set(delete_settings.allowed_columns)
if difference != set():
    raise InvalidQueryException(
        f"Invalid Columns {difference} to filter by, must be in {delete_settings.allowed_columns}"
    )
```

File: snuba/web/bulk_delete_query.py
Comment (onkar) [inline snuba/web/bulk_delete_query.py:183]:
```
return combine_or_conditions([_construct_condition(cond) for cond in conditions])
```

**Review by onkar — COMMENTED** [review]
Left comments. Please add unit tests since the existing coverage does not seem sufficient.

File: snuba/datasets/deletion_settings.py
Comment (kylemumma) [reply snuba/datasets/deletion_settings.py:6]: its looks like this function https://github.com/getsentry/snuba/blob/55398e54eaa9d96c4ce6a3eeb6816e41195cb975/snuba/web/delete_query.py#L137
will raise an exception and the delete will fail.

@MeredithAnya reading this function I realize it could cause an issue. Since we enforce max rows and fail in delete_from_table. if there are multiple tables the first one delete could succeed, but the second has too many rows and fails causing inconsistency. I think we need to change it so we check max rows on all tables before we start deleting on any.

File: snuba/web/bulk_delete_query.py
Comment (kylemumma) [reply snuba/web/bulk_delete_query.py:49]: I agree

File: snuba/datasets/deletion_settings.py
Comment (kylemumma) [reply snuba/datasets/deletion_settings.py:6]: I will hold off implementing the bug fix until this PR is merged since it would cause a merge conflict. I am happy to take on this change though.

File: snuba/web/bulk_delete_query.py
Comment (MeredithAnya) [reply snuba/web/bulk_delete_query.py:125]: @onkar I wondering if for now this is okay to keep whatever the error we want to know about it if we had a problem producing to the topic.

**Review by onkar — APPROVED** [review]

**Pull request merged**
